sdks/router-sdk/src/utils/encodeMixedRouteToPath.ts

```diff
@@ -29,32 +29,51 @@ export function encodeMixedRouteToPath(route: MixedRouteSDK<Currency, Currency>)
     path = [route.pathInput.isNative ? ADDRESS_ZERO : route.pathInput.address]
     types = ['address']
     let currencyIn = route.pathInput
+    let lastCurrencyOut = undefined
 
-    for (const pool of route.pools) {
+    for (let i = 0; i < route.pools.length; i++) {
+      const pool = route.pools[i]
```

Comment:
also this line is going to lead to errors some times I think, can you verify?

basically if we are operating on a route like this:
(dai-weth) (eth-usdc)

then currencyIn (which is the old currency out) may not equal either of the two tokens in the next pool

Reply:
mm I dont think thats true because `let currencyIn = route.pathInput`, which is a new variable for v4 that pre-calculates whether wrapping or unwrapping is needed

Reply:
but definitely worth double checking it works for all possible permutations of quotes